Packages/ClientRuntime/Sources/Waiters/Acceptor.swift

```diff
@@ -24,6 +24,18 @@ extension WaiterConfiguration {
             case failure
         }
 
+        /// Used as the root value of an `inputOutput` acceptor, which has the `input` and `output` fields
+        /// as its two top level properties.
+        public struct InputOutput {
+            public let input: Input?
+            public let output: Output?
```

Comment:
Do these need to be optional? The initializer requires non-nil values.

Reply:
These are optional to conform with the "every value is optional" assumption that is baked into JMESPath evaluation.

The initializer takes non-nil values because they will always actually be present when this type is instantiated; an `imputOutput` matcher always fails if there is no output.

Reply:
i really don't like this, but i see why it needs to be this way. 

it would be nice to not depend on the java jmes parser, or make changes to it, so that we aren't locked into behaviors like this. I noticed that its forcing our hand with how we generate the code.

Reply:
Can we add a comment here that adds this context since its not obvious, from just looking at the swift code, as to why these need to be optional

Reply:
Comment added.